### DISCUSSION TITLE
Improving the systemd unit file example

### DISCUSSION THREAD

**Discussion by l-x** [roadrunner-server/roadrunner#2069] [Ideas]
We recently came across the problem that `systemd stop roadrunner.service` with the unit file shown at https://docs.roadrunner.dev/docs/app-server/systemd sends both the Roadrunner process and the PHP worker processes SIGTERM at the same time. In our situation, this led to Roadrunner immediately restarting all workers and then waiting for them to finish - which of course never happened. After 30 seconds, Roadrunner then forcefully terminated these new workers.

We were able to solve this by setting the value for `KillMode` to `mixed` in the unit file. In our case, we also had to adjust the value for `TimeoutStopSec` to the Roadrunner configuration for `jobs.pool.destroy_timeout`.

Perhaps it makes sense to include these findings in the example unit file.

**Comment by rustatian** [comment]
Hey @l-x 👋🏻 
Yeah, great finding. Would you mind sending a PR to improve docs? (roadrunner-server/docs repo)

**Reply by l-x** [reply]
> If a service of Type=notify/Type=notify-reload sends "EXTEND_TIMEOUT_USEC=…", this may cause the stop time to be extended beyond TimeoutStopSec=. The first receipt of this message must occur before TimeoutStopSec= is exceeded, and once the stop time has extended beyond TimeoutStopSec=, the service manager will allow the service to continue to stop, provided the service repeats "EXTEND_TIMEOUT_USEC=…" within the interval specified, or terminates itself (see [sd_notify(3)](https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html#)).

How does Roadrunner behave in this situations? Maybe actually adjusting `TimeoutStopSec` is not required...

**Reply by rustatian** [reply]
RR does not handle these systemd events.
